### PR TITLE
Implement "your videos" tab

### PR DESCRIPTION
**Pull request by Mikescops**
This is a fix for request #277.

It implements a new "Your videos" tab and the ability to filter a bit more mime types when requesting photos.

image

I'd like some guidance for the needed translation.

Also there is a limitation on array merge that is fix only on PHP7.4 with the spread operator.

### CODE REVIEW

**Comment by Mikescops** [comment]
It is sightly different approach than #392, same limitations indeed.
Hope we can bring back the topic on the table.

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -50,6 +53,15 @@ class Application extends App implements IBootstrap {
 		// 'video/x-matroska'	// too rarely used for photos
 	];
 
+	/** Starting PHP7.4 we will be able to use spread operator (...) to merge the 2 arrays above */
+	public const MIMES = [
+		'image/png',
+		'image/jpeg',
+		'image/heic',
+		'video/mp4',
+		'video/quicktime',
```
Comment (skjnldsv) [inline lib/AppInfo/Application.php:62]:
```suggestion
		...IMAGE_MIMES,
		...VIDEO_MIMES,
```

Comment (skjnldsv) [reply lib/AppInfo/Application.php:62]: I thought it was js xD

File: src/router/index.js [hunk 2]
```diff
@@ -77,6 +77,15 @@ export default new Router({
 				showShared: true,
 			}),
 		},
+		{
+			path: '/videos',
+			component: Timeline,
+			name: 'videos',
+			props: route => ({
+				rootTitle: t('photos', 'Your videos'),
+				onlyVideos: true,
```
Comment (skjnldsv) [inline src/router/index.js:86]: we could also directly pass the mimes as props from here?

Comment (skjnldsv) [reply src/router/index.js:86]: Something like this ?
```suggestion
				mimesType: videoMimes,
```

File: lib/AppInfo/Application.php (hunk 1 above)
Comment (Mikescops) [reply lib/AppInfo/Application.php:62]: nope but as I said PHP implemented it in 7.4, that's awesome 👍

File: src/router/index.js (hunk 2 above)
Comment (Mikescops) [reply src/router/index.js:86]: good idea

Comment (skjnldsv) [reply src/router/index.js:86]: That way you can even do the spread from here:
```suggestion
				mimesType: [...videoMimes, ...imageMimes],
```

Comment (Mikescops) [reply src/router/index.js:86]: gotcha

**Comment by Mikescops** [comment]
Done @skjnldsv

**Review by skjnldsv — APPROVED** [review]
Code looks :ok_hand: :rocket:

**Comment by Mikescops** [comment]
Rebased with latest master.

**Comment by jancborchardt** [comment]
It’s still a bit strange to have the videos filtered if they show no previews at all. :\ Hope we still find a solution there, also ref the discussion from @jakobroehrl’s pull request in https://github.com/nextcloud/photos/pull/392

**Comment by Mikescops** [comment]
So I suggested an idea to @skjnldsv, when there is no preview, we display the name of the video in the tile. Also we can suggest with a banner a way to activate video previews

**Comment by jancborchardt** [comment]
> when there is no preview, we display the name of the video in the tile. Also we can suggest with a banner a way to activate video previews

Both nice enhancements! :) The banner should only show to admins though who can actually make the change.

**Comment by skjnldsv** [comment]
@Mikescops can you open issues for those? :)

**Comment by Mikescops** [comment]
#508

**Pull request merged**